### PR TITLE
Reflect read counter after marking all read

### LINKED ISSUE #7036: 'Mark all as read' for folder doesn't work
### Steps to reproduce

1. Open Mail App and ensure multiple messages are unread
2. Use context menu on folder and choose to "Mark all as read"
3. Verify if the messages have been marked as read

### Expected behavior

All messages should be marked as read and the number of unread emails should be 0

### Actual behavior

No emails are marked as read.

### Mail app version

1.13.8

### Mailserver or service

generic imap

### Operating system

Windows 10

### PHP engine version

PHP 8.1

### Web server

_No response_

### Database

_No response_

### Additional info

Using Nextcloud VM
Trying to mark all as read in my "Archive" folder.

### PR DESCRIPTION
**Pull request by hamza221**
fixes  #7036

### CODE REVIEW

**Comment by hamza221** [comment]
cc @MeIchthys

File: src/store/actions.js [hunk 1]
```diff
@@ -281,7 +281,14 @@ export default {
 				)
 			}
 
-			await markMailboxRead(mailboxId)
+			const updated = Object.assign({}, mailbox)
+			updated.unread = 0
+
+			await markMailboxRead(mailboxId).then(() => {
```
Comment (ChristophWurst) [inline src/store/actions.js:287]: move the *then* code to the next line. you are awaiting the return of the async function already

**Review by ChristophWurst — APPROVED** [review]
Looks good

@GretaD give this a test

**Comment by ChristophWurst** [comment]
needs backports too

**Review by GretaD — APPROVED** [review]
tested and works well

**Pull request merged**